src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/pre_compute.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "../../logging.hpp"
+#include "pre_compute.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/op/ops.hpp"
+#include "ov_ops/rotary_positional_embeddings.hpp"
+#include "transformations/common_optimizations/fuse_rotary_positional_embeddings.hpp"
+#include "openvino/pass/manager.hpp"
+
+namespace opp = ov::pass::pattern;
+
+namespace {
+    uint32_t align_to(uint32_t value, uint32_t alignment) {
+        return (value + alignment - 1) & ~(alignment - 1);
+    }
```

Comment:
I believe we already have one.. Maybe not in the common place though

Reply:
btw this not needed at all